sycl/include/sycl/detail/pi.h

```diff
@@ -648,6 +662,10 @@ constexpr pi_queue_properties PI_EXT_ONEAPI_QUEUE_FLAG_PRIORITY_LOW = (1 << 5);
 constexpr pi_queue_properties PI_EXT_ONEAPI_QUEUE_FLAG_PRIORITY_HIGH = (1 << 6);
 // clang-format on
 
+using pi_virtual_access_flags = pi_bitfield;
+constexpr pi_virtual_access_flags PI_VIRTUAL_ACCESS_FLAG_RW = (1 << 0);
+constexpr pi_virtual_access_flags PI_VIRTUAL_ACCESS_FLAG_READ_ONLY = (1 << 2);
```

Comment:
These seem to be a subset of [ur_mem_flags_t](https://oneapi-src.github.io/unified-runtime/core/api.html#ur-mem-flags-t), could potentially reuse those enumerations with some valid usage language in the spec to disallow the rest in this context. A separate enum is probably cleaner though.

Reply:
I am no opposed to consolidating them for UR and just specify the restrictions in the access functions. That said, we do run the risk of locking the interface in a way that makes it hard and/or awkward to adjust in case new access modes are added in the future.

Reply:
As [here](https://github.com/intel/llvm/pull/8954#discussion_r1184885790) I can't resolve this discussion but happy with it as is 👍